supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkJobExecutionActorSpec.scala

```diff
@@ -0,0 +1,560 @@
+
+  private val mockFailedHttpResponse = HttpResponse(StatusCodes.OK, HttpEntity(ContentTypes.`application/json`, mockFailedClusterResponse.toJson.toString))
+
+  private val mockBadHttpResponse = HttpResponse(StatusCodes.BadRequest)
+
+  private val sampleSubmissionResponse =
+    """
+      |Running Spark using the REST application submission protocol.
+      |16/08/06 18:35:26 INFO rest.RestSubmissionClient: Submitting a request to launch an application in spark://host-10-0-1-53:6066.
+      |16/08/06 18:35:27 INFO rest.RestSubmissionClient: Submission successfully created as driver-20160806183527-0006. Polling submission state...
+      |16/08/06 18:35:27 INFO rest.RestSubmissionClient: Submitting a request for the status of submission driver-20160806183527-0006 in spark://host-10-0-1-53:6066.
+      |16/08/06 18:35:27 INFO rest.RestSubmissionClient: State of driver driver-20160806183527-0006 is now RUNNING.
+      |16/08/06 18:35:27 INFO rest.RestSubmissionClient: Driver is running on worker worker-20160801162431-10.0.1.55-43834 at 10.0.1.55:43834.
+      |16/08/06 18:35:27 INFO rest.RestSubmissionClient: Server responded with CreateSubmissionResponse:
+      |{
+      |  "action" : "CreateSubmissionResponse",
+      |  "message" : "Driver successfully submitted as driver-20160806183527-0006",
+      |  "serverSparkVersion" : "1.6.1",
+      |  "submissionId" : "driver-20160806183527-0006",
+      |  "success" : true
+      |}
+    """.stripMargin
+
+  private val failedOnStderr =
+    """
+      |runtime {
+      | appMainClass: "test"
+      | failOnStderr: true
+      |}
+    """.stripMargin
+
+  private val passOnStderr =
+    """
+      |runtime {
+      | appMainClass: "test"
+      | failOnStderr: false
+      |}
+    """.stripMargin
+
+  private val backendClusterConfig = ConfigFactory.parseString(
+    s"""{
+        |  root = "local-cromwell-executions"
+        |  filesystems {
+        |    local {
+        |      localization = [
+        |        "hard-link", "soft-link", "copy"
+        |      ]
+        |    }
+        |  }
+        |  master: "spark"
+        |  deployMode: "cluster"
+        |}
+        """.stripMargin)
+
+  private val backendClientConfig = ConfigFactory.parseString(
+    s"""{
+        |  root = "local-cromwell-executions"
+        |  filesystems {
+        |    local {
+        |      localization = [
+        |        "hard-link", "soft-link", "copy"
+        |      ]
+        |    }
+        |  }
+        |  master: "local"
+        |  deployMode: "client"
+        |}
+        """.stripMargin)
+
+  private val timeout = Timeout(20.seconds)
+
+  after {
+    Mockito.reset(sparkProcess)
+    Mockito.reset(sparkClusterProcess)
+  }
+
+  before {
+    when(sparkClusterProcess.logger).thenReturn(Logger(LoggerFactory.getLogger(getClass.getName)))
+    when(sparkClusterProcess.completeMonitoringProcess(any[Path], any[String], any[Promise[Unit]])).thenCallRealMethod()
```

Comment:
Ok, I have changed `SparkExecutionActorSpec` only unit tests and `SparkClusterProcessSpec` as separate unit test. However `SparkClusterProcess` needs not only EC (I forgot to mention before) but the `actorsystem` because of Spray client for making rest calls.